sw/host/opentitantool/src/command/console.rs

```diff
@@ -0,0 +1,246 @@
+use anyhow::{anyhow, Result};
+use erased_serde::Serialize;
+use nix::unistd::isatty;
+use raw_tty::TtyModeGuard;
+use regex::Regex;
+use std::fs::File;
+use std::io;
+use std::io::{ErrorKind, Read, Write};
+use std::os::unix::io::AsRawFd;
+use std::time::{Duration, Instant};
+use structopt::StructOpt;
+
+use crate::command::CommandDispatch;
+use opentitanlib::io::uart::Uart;
+use opentitanlib::transport::{Capability, Transport};
+use opentitanlib::util::file;
+
+#[derive(Debug, StructOpt)]
+pub struct Console {
+    #[structopt(short, long, help = "Do not print console start end exit messages.")]
+    quiet: bool,
+
+    #[structopt(short, long, help = "Log console output to a file")]
+    logfile: Option<String>,
+
+    #[structopt(short, long, help = "Exit after a timeout in seconds.")]
```

Comment:
I really wish there was a sensible way to just parse durations. I'd much prefer `--timeout 10s`. But meh.